### PR TITLE
Automate aeglos prices update

### PR DESCRIPTION
**Pull request by yvesleconte**

### CODE REVIEW

File: automate_prices_update_aeglos.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+cg = CoinGeckoAPI()
+
+
+def get_sym_aeglos():
+    """
+    :return: a list, the syms of the investments made in Aeglos without duplicates
+    """
+    syms = pd.read_excel(FILE_PATH + FILE_NAME_AEGLOS, sheet_name='MtM Info')['Sym'].to_list()
+    syms_no_duplicates = []
+    for sym in range(len(syms)):
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:17]: Do not do a loop, use `list(set(syms))`

File: automate_prices_update_aeglos.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+FILE_NAME_AEGLOS = 'Aeglos.xlsx'
+
+cg = CoinGeckoAPI()
+
+
+def get_sym_aeglos():
+    """
+    :return: a list, the syms of the investments made in Aeglos without duplicates
+    """
+    syms = pd.read_excel(FILE_PATH + FILE_NAME_AEGLOS, sheet_name='MtM Info')['Sym'].to_list()
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:15]: rebase against `main` branch and use functions in `src/aeglos/read_write.py`

File: automate_prices_update_aeglos.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            del prices[coin['id']]
+    return prices
+
+
+def write_mtm_prices_aeglos(prices):
+    """
+    :param prices: dictionary, a nested dictionary of the prices in USD of each sym
+    :return: writes the prices in the MtM Info sheet of Aeglos
+    """
+    wb = load_workbook(FILE_PATH + FILE_NAME_AEGLOS)
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:49]: Do the following:
- load the whole sheet of mtm info using the function in `src/aeglos/read_write.py`
- replace the price column with your prices
- write the whole df to the mtm sheet using the function in `src/aeglos/read_write.py`

This allows to have that whole read/write logic being taken care of externally with the local/remote options and you just need to replace the prices in the df

The way I would do it is that you have:
1. a df mtm info
2. a df with sym and price column
3. Then you just join on key = sym

File: automate_prices_update_aeglos.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        if coin['symbol'] in lower_sym:
+            ids.append(coin['id'])
+    ids.remove('ethereum-wormhole')
+    ids.remove('flower')
+    prices = cg.get_price(ids=ids, vs_currencies='usd')
+    for coin in coins_list:
+        if coin['id'] in prices:
+            prices[coin['symbol'].upper()] = prices[coin['id']]
+            del prices[coin['id']]
+    return prices
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:41]: I think getting a df is cleaner than a dict and you can reuse it more easily if needs be
`pd.DataFrame(prices)).transpose().reset_index().rename({"index":"sym","usd":"price"})`

File: automate_prices_update_aeglos.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    :param syms: list, the syms of the investments made in Aeglos without duplicates, in uppercase
+    :return: dict, the mtm price of each sym in Aeglos, beware of extra ids due to symbol correlation
+    """
+    coins_list = cg.get_coins_list()
+    ids = []
+    lower_sym = [sym.lower() for sym in syms]
+    for coin in coins_list:
+        if coin['symbol'] in lower_sym:
+            ids.append(coin['id'])
+    ids.remove('ethereum-wormhole')
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:34]: please add a comment as to why you are doing this for future ref

File: automate_prices_update_aeglos.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    ids.remove('flower')
+    prices = cg.get_price(ids=ids, vs_currencies='usd')
+    for coin in coins_list:
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:37]: are you sure this for loop is the cleaner and easier way to do it?

File: automate_prices_update_aeglos.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import pandas as pd
+from openpyxl import load_workbook
+from pycoingecko import CoinGeckoAPI
+
+FILE_PATH = 'C:/dev/data/'
+FILE_NAME_AEGLOS = 'Aeglos.xlsx'
+
+cg = CoinGeckoAPI()
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:8]: please add it in the function, we don't want to init it if there is no need

File: automate_prices_update_aeglos.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
+import pandas as pd
+from openpyxl import load_workbook
+from pycoingecko import CoinGeckoAPI
+
+FILE_PATH = 'C:/dev/data/'
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:5]: re use the constants in `constants.py`

File: automate_prices_update_aeglos.py (hunk 1 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:17]: Done

File: automate_prices_update_aeglos.py (hunk 2 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:15]: Done

File: automate_prices_update_aeglos.py (hunk 7 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:8]: Done

File: automate_prices_update_aeglos.py (hunk 8 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:5]: Done

File: automate_prices_update_aeglos.py (hunk 4 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:41]: Done

File: automate_prices_update_aeglos.py (hunk 5 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:34]: Done

File: automate_prices_update_aeglos.py (hunk 3 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:49]: Done

File: automate_prices_update_aeglos.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+            ids.append(coin['id'])
+    #remove other existing coins with same symbols as coins in Aeglos.py
+    ids.remove('ethereum-wormhole')
+    ids.remove('flower')
+    prices = cg.get_price(ids=ids, vs_currencies='usd')
+    for coin in coins_list:
+        if coin['id'] in prices:
+            prices[coin['symbol'].upper()] = prices[coin['id']]
+            del prices[coin['id']]
+    prices = (pd.DataFrame(prices)).transpose().reset_index().rename({"index": "sym", "usd": "price"})
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:36]: superfluous parenthesis?

File: automate_prices_update_aeglos.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def write_mtm_prices_aeglos(prices):
+    """
+    :param prices: dataframe, the prices in USD of each sym
+    :return: writes the prices in the MtM Info sheet of Aeglos
+    """
+    ws = get_mtm_info_aeglos()
+    for row in range(ws.shape[0]):
+        ws.iloc[row, 4] = prices[prices['index'].str.contains(ws.iloc[row, 2])].iloc[0, 1]
```
[the remark this answers is not in the document]
Comment (Arthurim) [reply automate_prices_update_aeglos.py:47]: + df operations > loops

File: automate_prices_update_aeglos.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+def get_sym_aeglos():
+    """
+    :return: list, the syms of the investments made in Aeglos without duplicates
+    """
+    syms = get_mtm_info_aeglos()['Sym'].to_list()
+    syms_no_duplicates = list(set(syms))
+    return syms_no_duplicates
+
+
+def get_mtm_prices_aeglos(syms):
```
[the remark this answers is not in the document]
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:16]: Ok I get it, good job ! I didn't see it this way

File: automate_prices_update_aeglos.py (hunk 9 above)
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:36]: Definitely.

File: automate_prices_update_aeglos.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (yvesleconte) [reply automate_prices_update_aeglos.py:16]: However, syms and ids do not necessary match together in syms_to_ids in your code

File: automate_prices_update_aeglos.py [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    prices.columns = ['sym', 'price']
+    return prices
+
+
+def write_mtm_prices_aeglos(prices):
+    """
+    :param prices: dataframe, the prices in USD of each sym
+    :return: writes the prices in the MtM Info sheet of Aeglos
+    """
+    mtm_infos = get_mtm_info_aeglos()
```
Comment (Arthurim) [inline automate_prices_update_aeglos.py:49]:
```
    mtm_infos = get_mtm_info_aeglos()
    mtm_infos_updated = pd.merge(mtm_infos.drop("MtM Price",axis=1),prices.rename(columns={"sym":"Sym", "price":"MtM Price"}), on='Sym', how='left')
    mtm_infos_updated["MtM Value"] =  mtm_infos_updated["MtM Amount"] * mtm_infos_updated["MtM Price"]
    mtm_infos_updated[["ExecutionId", "Protocol", "Sym", "MtM Amount", "MtM Price","MtM Value"]]
```

**Pull request merged**